`ConcreteBuffer::nbytes` goes wrong

[`test_buffer.py::ConcreteBufferBasicTC::test_ConcreteBuffer_from_ndarray`](https://github.com/solvcon/modmesh/blob/ff4b4d654ab9abc53fe4096bf952e67929f972ac/tests/test_buffer.py#L80) fails after PR #363 is merged.  See CI failure for more information at https://github.com/solvcon/modmesh/actions/runs/9537773849/job/26290385018#step:12:98 :

```
  ..\..\..\..\tests\test_buffer.py::ConcreteBufferBasicTC::test_ConcreteBuffer_from_ndarray FAILED [  1%]
  
  ================================== FAILURES ===================================
  ___________ ConcreteBufferBasicTC.test_ConcreteBuffer_from_ndarray ____________
  
  self = <test_buffer.ConcreteBufferBasicTC testMethod=test_ConcreteBuffer_from_ndarray>
  
      def test_ConcreteBuffer_from_ndarray(self):
      
          buf = modmesh.ConcreteBuffer(24)
          self.assertFalse(buf.is_from_python)
      
          ndarr = np.arange(24, dtype='float64').reshape((2, 3, 4))
      
          buf = modmesh.ConcreteBuffer(array=ndarr)
  >       self.assertEqual(ndarr.nbytes, buf.nbytes)
  E       AssertionError: 192 != 0
  
  ..\..\..\..\tests\test_buffer.py:80: AssertionError
```

The failure only happens on Windows, but I guess it could be a random behavior.

---

Ref issue #305

--- a/cpp/modmesh/buffer/BufferBase.hpp
+++ b/cpp/modmesh/buffer/BufferBase.hpp
@@ -96,7 +96,9 @@ protected:
 
     BufferBase(int8_t * start, int8_t * end)
         : m_begin(start)
-        , m_end(end){};
+        , m_end(end)
+    {
+    }
 
     BufferBase(BufferBase const &) = delete;
     BufferBase(BufferBase &&) = delete;

--- a/cpp/modmesh/buffer/BufferExpander.hpp
+++ b/cpp/modmesh/buffer/BufferExpander.hpp
@@ -75,7 +75,7 @@ public:
         : BufferBase<BufferExpander>() // don't delegate m_begin and m_end, which will be overwritten later
         , m_concrete_buffer(clone ? buf->clone() : buf)
     {
-        m_begin = m_concrete_buffer->data(); // override m_begin and m_end once we have the data
+        m_begin = m_concrete_buffer->data(); // overwrite m_begin and m_end once we have the data
         m_end = m_begin + m_concrete_buffer->size();
         m_end_cap = m_begin + m_concrete_buffer->size();
     }
@@ -83,7 +83,7 @@ public:
     BufferExpander(size_type nbyte, ctor_passkey const &)
         : BufferBase<BufferExpander>(nullptr, nullptr) // initialize m_begin and m_end to nullptr, which will be overwritten later
     {
-        expand(nbyte); // override m_begin and m_end once we have the data
+        expand(nbyte); // overwrite m_begin and m_end once we have the data
     }
 
     BufferExpander(ctor_passkey const &)

--- a/cpp/modmesh/buffer/ConcreteBuffer.hpp
+++ b/cpp/modmesh/buffer/ConcreteBuffer.hpp
@@ -174,7 +174,7 @@ public:
         , m_nbytes(nbytes)
         , m_data(allocate(nbytes))
     {
-        m_begin = m_data.get(); // override m_begin and m_end once we have the data
+        m_begin = m_data.get(); // overwrite m_begin and m_end once we have the data
         m_end = m_begin + m_nbytes;
     }
 
@@ -193,7 +193,7 @@ public:
         , m_nbytes(nbytes)
         , m_data(data, data_deleter_type(std::move(remover)))
     {
-        m_begin = m_data.get(); // override m_begin and m_end once we have the data
+        m_begin = m_data.get(); // overwrite m_begin and m_end once we have the data
         m_end = m_begin + m_nbytes;
     }
 
@@ -213,14 +213,13 @@ public:
         , m_nbytes(other.m_nbytes)
         , m_data(allocate(other.m_nbytes))
     {
+        m_begin = m_data.get(); // overwrite m_begin and m_end once we have the data
+        m_end = m_begin + m_nbytes;
         if (size() != other.size())
         {
             throw std::out_of_range("Buffer size mismatch");
         }
         std::copy_n(other.data(), size(), data());
-
-        m_begin = m_data.get(); // override m_begin and m_end once we have the data
-        m_end = m_begin + m_nbytes;
     }
 #ifdef __GNUC__
 #pragma GCC diagnostic pop

